### DISCUSSION TITLE
Is there a way to redirect previous url and keep state entered?

### DISCUSSION THREAD

**Discussion by waldobin** [http-rs/tide#839] [Q&A]
Hi there,

For example,
- I filled up register form with username, first & last name and password..., then click on Submit button
- The tide received these parameters, then will need to check length of username, assume that it is not valid. Then I want to redirect the register form again with keeping a previous state filled up.

Thanks all

**Comment by Fishrock123** [comment]
Consider using some form of sessions: https://github.com/http-rs/async-session

**Reply by jbr** [reply]
this probably should link to [tide::sessions](https://docs.rs/tide/latest/tide/sessions/index.html) — you'd want to choose a session store and put the username/first/last data into the session and retrieve it on the next request
